### PR TITLE
Xcmetrics build details

### PR DESCRIPTION
**Pull request by ngranander**
## Hey, I just made a Pull Request!

This adds components for viewing detailed information of builds. Details includes data such as build host specifications, errors, warnings, metadata and a timeline for each build target.

The main component is `BuildDetailsComponent` and it can be viewed from the builds table (by clicking on the arrow on the left hand side of a row) or by navigating to `/xcmetrics/build/:id`. The idea behind having it in the table is to enable quick and easy exploration of multiple builds.

#### Build details in build table:

Screenshot 2021-08-04 at 19 12 32

#### Build details as a page when navigating to `/xcmetrics/builds/:id`

Screenshot 2021-08-04 at 19 14 13

#### Expanded build error details. The errors in the right hand side panels are truncated to save space. If the user needs all information they can simple click the message and see this dialog with the full detail. Warnings work in the same way.

Screenshot 2021-08-04 at 19 14 45

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [X] Tests for new functionality and regression tests for bug fixes
- [X] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by freben** [comment]
Hey! I have it on my radar to review this, probably tomorrow. Just wanted to mention in the mean time that there's a small conflict that has crept in.

**Comment by ngranander** [comment]
Thanks for notifying me @freben! I will wait for #6793 and fix the conflict 👍

**Comment by freben** [comment]
Merged the fix!

**Review by freben — COMMENTED** [review]
This is looking awesome! Just some small comments.

File: plugins/xcmetrics/src/components/BuildDetailsComponent/BuildDetailsComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+import { Progress, StructuredMetadataTable } from '@backstage/core-components';
+import { Alert } from '@material-ui/lab';
+import { useAsync } from 'react-use';
+import { useApi } from '@backstage/core-plugin-api';
+import { formatDuration, formatStatus, formatTime } from '../../utils';
+import { StatusIconComponent as StatusIcon } from '../StatusIconComponent';
+import { BackstageTheme } from '@backstage/theme';
+import { AccordionComponent } from '../AccordionComponent';
+import { BuildTimelineComponent } from '../BuildTimelineComponent';
+import { PreformattedTextComponent } from '../PreformattedTextComponent';
```
[the remark this answers is not in the document]
Comment (ngranander) [reply plugins/xcmetrics/src/components/BuildDetailsComponent/BuildDetailsComponent.tsx:28]: Yeah, I actually totally agree! I don't really know why I went for this convention, probably saw it somewhere and it stuck 😄. Would it be ok to change all of these (and the named exports in index files) in this PR or would you prefer that I create separate PRs for that?

[the remark this answers is not in the document]
Comment (freben) [reply plugins/xcmetrics/src/components/BuildDetailsComponent/BuildDetailsComponent.tsx:28]: They are not exported right? No need to rush, you can do it separately at any point later, if you have the time.

File: plugins/xcmetrics/src/api/XcmetricsClient.ts [hunk 2]
```diff
@@ -65,80 +58,88 @@ export class XcmetricsClient implements XcmetricsApi {
...
+        .endOf('day')
+        .startOf('second')
+        .toISO({ suppressMilliseconds: true }),
+      status: filters.buildStatus,
+      projectName: filters.project,
+      page,
+      per: perPage,
     });
 
-    if (!response.ok) {
```
[the remark this answers is not in the document]
Comment (ngranander) [reply plugins/xcmetrics/src/api/XcmetricsClient.ts:87]: I moved that logic to the new private methods `get` and `post` just to be a little less repetitive in each method. So if I'm not mistaken, the behavior should still be the same :)

File: plugins/xcmetrics/src/components/DatePickerComponent/DatePickerComponent.tsx [hunk 3]
```diff
@@ -62,6 +62,7 @@ export const DatePickerComponent = ({
 }: InputProps & DatePickerProps) => (
   <>
     <Typography variant="button">{label}</Typography>
+    <br />
```
[the remark this answers is not in the document]
Comment (ngranander) [reply plugins/xcmetrics/src/components/DatePickerComponent/DatePickerComponent.tsx]: Unfortunately `paragraph` introduces some unwanted margin, but I got rid of the `<br/>` and used the same approach as in `Select` (which is the component I want to imitate)

File: plugins/xcmetrics/src/api/XcmetricsClient.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (freben) [reply plugins/xcmetrics/src/api/XcmetricsClient.ts:87]: oh that explains it, i'm just blind!

**Review by freben — APPROVED** [review]
Looking goooood

**Pull request merged**